app/views.py

```diff
@@ -86,10 +86,68 @@ def get_context_data(self, **kwargs):
         return context
 
 
-class EditProfileView(LoginRequiredMixin, UpdateView):
+class SocialLinkFormMixin(FormMixin):
+    def get_class_name(self):
+        return (
+            self.model if self.model else self.object.__class__ if self.object else self.classname
+        )
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+
+        context["has_social_links_form"] = True
+
+        initial_data = {
+            "object_id": 0 if not self.object else self.object.pk,
+            "content_type": ContentType.objects.get_for_model(self.get_class_name()),
+        }
+        queryset = (
+            SocialLinkAttachement.objects.none()
+            if not self.object
+            else SocialLinkAttachement.objects.filter(
+                object_id=initial_data["object_id"],
+                content_type=initial_data["content_type"],
+            )
+        )
+
+        context["formset"] = (
+            kwargs.pop("social_formset")
+            if "social_formset" in kwargs
+            else SocialLinkFormSet(
+                prefix="social_links",
+                queryset=queryset,
+            )
+        )
+
+        SocialLinkFormSet.form.base_fields["content_type"].initial = initial_data["content_type"]
+        SocialLinkFormSet.form.base_fields["object_id"].initial = initial_data["object_id"]
```

Comment:
Is there a better way to pass initial data to these fields? Passing an `initial` argument into `SocialLinkFormSet` does not pass the values into `empty_form`.